api/uniter/relation.go

```diff
@@ -114,18 +112,16 @@ func (r *Relation) Endpoint() (*Endpoint, error) {
 	return &Endpoint{r.toCharmRelation(result.Endpoint.Relation)}, nil
 }
 
-// Unit returns a RelationUnit for the supplied unit.
-func (r *Relation) Unit(u *Unit) (*RelationUnit, error) {
-	if u == nil {
-		return nil, fmt.Errorf("unit is nil")
-	}
-	result, err := r.st.relation(r.tag, u.tag)
+// Unit returns a RelationUnit for the supplied unitTag and applicationTag.
+func (r *Relation) Unit(uTag names.UnitTag, aTag names.ApplicationTag) (*RelationUnit, error) {
```

Comment:
This should really be Unit(uTag names.UnitTag).  We can get the application tag easily enough via the names pkg.